source/material_model/visco_plastic.cc

```diff
@@ -42,6 +42,27 @@ namespace aspect
       }
     }
 
+    template <int dim>
+    bool
+    ViscoPlastic<dim>::
+    get_plastic_yielding (const double &pressure,
+                          const double &temperature,
+                          const std::vector<double> &composition,
+                          const SymmetricTensor<2,dim> &strain_rate) const
+    {
+      bool plastic_yielding = false;
+
+      const std::vector<double> volume_fractions = MaterialUtilities::compute_volume_fractions(composition, get_volumetric_composition_mask());
+
+      const std::pair<std::vector<double>, std::vector<bool> > calculate_viscosities =
+        calculate_isostrain_viscosities(volume_fractions, pressure, temperature, composition, strain_rate, viscous_flow_law, yield_mechanism);
+
+      std::vector<double>::const_iterator max_composition = std::max_element(volume_fractions.begin(),volume_fractions.end());
+      plastic_yielding = calculate_viscosities.second[std::distance(volume_fractions.begin(),max_composition)];
```

Comment:
So this function in reality returns whether the material with the highest volume fraction is plastically yielding? That definitely needs to go into the documentation. Is there a reason you do not check if *any* composition is yielding?

Reply:
This bit is a repetition of some lines in `evaluate`, where I did provide some comments:
` // Decide based on the maximum composition if material is yielding.
   // This avoids for example division by zero for harmonic averaging (as plastic_yielding
   // holds values that are either 0 or 1), but might not be consistent with the viscosity
  // averaging chosen.`

Reply:
I agree that choosing the maximum composition is the right approach. If many of the functions in visco_plastic are moved to MaterialUtilities, then hopefully things can be consistent between the compositional field and particle approach.